### PR TITLE
ci: make github action effective for release-5.1

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: HunDunDM <hundundm@gmail.com>

### What is changed and how it works?

ci: make github action effective for `release-5.1`

### Check List

Tests

- No code

Code changes

Side effects

Related changes

- Need to cherry-pick to the release branch

### Release note

```release-note
None
```

### CODE REVIEW

File: .github/workflows/check-dashboard.yaml [hunk 1]
```diff
@@ -4,13 +4,15 @@ on:
...
       - release-5.0
+      - release-5.1
     paths:
       - "go.mod"
   pull_request:
     branches:
       - master
       - release-4.0
       - release-5.0
+      - release-5.1
```
Comment (disksing) [inline .github/workflows/check-dashboard.yaml:15]: how about `release-5.*`

Comment (disksing) [reply .github/workflows/check-dashboard.yaml:15]: or maybe we can use `release-*` and ignore `release-[123]*`

Comment (disksing) [reply .github/workflows/check-dashboard.yaml:15]: examples: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-branches-and-tags

Comment (HunDunDM) [reply .github/workflows/check-dashboard.yaml]: They do not run in `release-3.1` or lower.

**Review by disksing — APPROVED** [review]

**Pull request merged**